apps/hash-ai-worker-ts/src/activities/shared/get-llm-response.ts

```diff
@@ -0,0 +1,467 @@
+import Ajv from "ajv";
+import dedent from "dedent";
+import type OpenAI from "openai";
+import type {
+  ChatCompletion,
+  ChatCompletion as OpenAiChatCompletion,
+  ChatCompletionCreateParamsNonStreaming,
+  FunctionParameters as OpenAiFunctionParameters,
+} from "openai/resources";
+import { promptTokensEstimate } from "openai-chat-tokens";
+
+import { logger } from "../../shared/logger";
+import type {
+  AnthropicMessagesCreateParams,
+  AnthropicToolDefinition,
+  MessageContent,
+} from "./get-llm-response/anthropic-client";
+import {
+  anthropicMessageModelToMaxOutput,
+  createAnthropicMessagesWithTools,
+  isAnthropicContentToolUseContent,
+} from "./get-llm-response/anthropic-client";
+import type {
+  AnthropicLlmParams,
+  AnthropicResponse,
+  LlmParams,
+  LlmResponse,
+  LlmStopReason,
+  LlmToolDefinition,
+  OpenAiLlmParams,
+  ParsedLlmToolCall,
+} from "./get-llm-response/types";
+import { isLlmParamsAnthropicLlmParams } from "./get-llm-response/types";
+import { modelToContextWindow, openai } from "./openai-client";
+import { stringify } from "./stringify";
+
+const mapLlmToolDefinitionToAnthropicToolDefinition = (
+  tool: LlmToolDefinition,
+): AnthropicToolDefinition => ({
+  name: tool.name,
+  description: tool.description,
+  input_schema: tool.inputSchema,
+});
+
+const mapLlmToolDefinitionToOpenAiToolDefinition = (
+  tool: LlmToolDefinition,
+): OpenAI.Chat.Completions.ChatCompletionTool => ({
+  type: "function",
+  function: {
+    name: tool.name,
+    parameters: tool.inputSchema as OpenAiFunctionParameters,
+    description: tool.description,
+  },
+});
+
+const parseToolCallsFromAnthropicResponse = (
+  response: AnthropicResponse,
+): ParsedLlmToolCall[] =>
+  response.content
+    .filter(isAnthropicContentToolUseContent)
+    .map(({ id, name, input }) => ({ id, name, input }));
+
+const mapOpenAiFinishReasonToLlmStopReason = (
+  finishReason: OpenAiChatCompletion["choices"][0]["finish_reason"],
+): LlmStopReason => {
+  switch (finishReason) {
+    case "stop":
+      return "stop";
+    case "tool_calls":
+      return "tool_use";
+    case "length":
+      return "length";
+    case "content_filter":
+      return "content_filter";
+    default:
+      throw new Error(`Unexpected OpenAi finish reason: ${finishReason}`);
```

Comment:
```suggestion
      throw new Error(`Unexpected OpenAI finish reason: ${finishReason}`);
```

Reply:
Thanks, addressed in https://github.com/hashintel/hash/pull/4318/commits/f97fda48b5855f1725daa645196a3da68373818b